staging/src/k8s.io/legacy-cloud-providers/aws/aws.go

```diff
@@ -4609,6 +4623,19 @@ func (c *Cloud) getFullInstance(nodeName types.NodeName) (*awsInstance, *ec2.Ins
 	return awsInstance, instance, err
 }
 
+func (c *Cloud) nodeNameToProviderID(nodeName types.NodeName) (InstanceID, error) {
+	// TODO: get node from cache
+	node, err := c.kubeClient.CoreV1().Nodes().Get(string(nodeName), metav1.GetOptions{})
```

Comment:
@zhan849, can you please check for empty value and fall back to `findInstanceByNodeName` without ugly error about failed `MapToAWSInstanceID`?

Reply:
done. Added a few checks to fail early when nodeName is empty or providerID is not set yet.